src/appleseed/renderer/modeling/light/light.cpp

```diff
@@ -118,9 +113,13 @@ bool Light::on_frame_begin(
     if (!ConnectableEntity::on_frame_begin(project, parent, recorder, abort_switch))
         return false;
 
-    m_flags = 0;
-    if (is_light_tree_compatible())
+    if (m_flags & LightTreeCompatible)
```

Comment:
Ah, that's another approach indeed. The downside is that it won't work if we add another flag.

What I proposed is different, see the description on Slack.